src/test_l1_dsAudio.c

```diff
@@ -5611,31 +7062,46 @@ void test_l1_dsAudio_positive_dsEnableLEConfig(void) {
 	gTestID = 89; // Assuming Test Case ID is 89
 	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
-	int result;
-	intptr_t  handle[NUM_OF_PORTS] = {INT_ARRAY_INIT};
+        int result;
+        intptr_t* handle = NULL;
 
-	// Step 01: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // allocate memory for the handles
+        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
+        if(handle == NULL)
+        {
+                UT_LOG_ERROR("Failed to allcoate memory for handlers");
+                return;
+        }
 
-	// Steps 02 to 03: Enable LE Config for each port
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		result = dsGetAudioPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(handle[i], null_handle);
+        // Step 01: Initialize audio ports
+        result = dsAudioPortInit();
+        UT_ASSERT_EQUAL(result, dsERR_NONE);
 
-		// Enable LE Config
-		result = dsEnableLEConfig(handle[i], true);
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
+        // Steps 02 to 03: Enable LE Config for each port
+        for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+                result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
 
-	}
+                // Enable LE Config
+                result = dsEnableLEConfig(*(handle+i), true);
+             if((gSourceType == 0) && (gDSAudioPortConfiguration[i].ms12_capabilites & 0x400)) {
```

Comment:
While clean the code please take care replace 0x400 to corresponding enum values